src/languages/yaml.js

```diff
@@ -72,11 +76,11 @@ export default function(hljs) {
       },
       { // local tags
         className: 'type',
-        begin: '!' + hljs.UNDERSCORE_IDENT_RE,
+        begin: '!' + YAML_TAG_RE,
```

Comment:
Let's change it to YAML_TAG_PREFIX - unless you tell me I'm mistaken? And then I think this looks ok.

Reply:
Tag prefixes are available in YAML but I'm not sure they work exactly the way you think they do. Basically, at the top of a YAML document you can define a tag prefix which is basically just a shorthand for a longer tag, e.g.:
```
%TAG !e! tag:example.com,2000:app/
---
!e!foo "bar"
```
Which converts `!e!foo` to `!<tag:example.com,2000:app/foo>` during parsing, according to the spec.

Reply:
I'm open to other names for this variable, but I'm not sure `YAML_TAG_PREFIX` is appropriate, since we're matching the whole tag.

Reply:
https://yaml.org/spec/1.2/spec.html#id2783273

I'm thinking "local tag prefix" which start with `!` from my read but globals don't have `!` evidentaly.  Is your intention here that this is matching tags as well as tag prefixes, yes?

Reply:
Ie, we're matching `!tagname!` with this also, yes?

Reply:
After reading more carefully, I don't think it works the way I thought it did either. Most of what I've read so far merely relates to the `%TAG` directive, which allows three tag prefixes, !, !!, and ! \w+ !, then it allows a replacement that includes an URI characters.

However, this doesn't define what is allowed in full tags. I actually can't find what is allowed in full tags in this document, but given how PyYAML uses the tags, it seems they can include at least `/` and `:` and `.`

Reply:
I propose allowing all URI characters in tags since this is how other syntax highlighters are doing it (e.g. vim, although I don't think vim allows square brackets).

Reply:
> However, this doesn't define what is allowed in full tags.

What are you calling a "full tag" here?

A tag is only a handle and a prefix unless I'm missing something.

<img width="519" alt="Screen Shot 2020-04-17 at 5 18 14 AM" src="https://user-images.githubusercontent.com/6473/79553699-e11ec400-806a-11ea-97e1-286387692d40.png">

Then there are primary handles `!`, secondary `!!` and named handles:

```
[92] | c-named-tag-handle | ::= | “!” ns-word-char+ “!”
```

Isn't that it?  I feel like this rule (as currently written) is also going to capture named tag handles - and that might be ok... if not we should perhaps add matchers for the 3 tag handle variants.

Reply:
If all this extra complexity is to deal with tag handles vs prefixes would it be simpler to have a sep rule for the handles? entirely? ` !, !!, and ! \w+ !` as you already said. Would that simplify the prefix rule?

Reply:
To clarify I by "full tag" I meant `!yaml!str` where the tag prefix (or tag handle) refers to `!yaml!`. What is allowed to come after the tag prefix (in this example: `str`) is not very clearly defined in the spec. The only thing I can find is in section `3.2.1.2 Tags` and it goes like this:

`local tags are restricted to the URI character set and use URI character escaping.`

And it goes on to describe how different languages will use the tags differently (e.g. Java tags use `.`). So full URI character set is what I recommend for tags (the part after the prefix). If you want to do it entirely properly, probably what would be required is three rules for the three valid prefixes, plus excluding exclamation points from the character set.